boards/native/Makefile.include

```diff
@@ -97,15 +94,15 @@ CXXUWFLAGS +=
 CXXEXFLAGS +=
 
 ifeq ($(OS_ARCH),x86_64)
-  ifeq ($(NATIVE_64BIT), 1)
+  ifeq ($(BOARD),native64)
     LINKFLAGS += -m64
   else
     LINKFLAGS += -m32
   endif
```

Comment:
Would be cleaner to move this directly to `boards/native64`

Reply:
I can do that. The question is how, since I can't just override `LINKFLAGS` and `CFLAGS`.

1. Move the logic from `boards/native` to something like `boards/common/native` and set the flags in `native` and `native64`.
2. Do something like `ifeq ($(BOARD),native) LINKFLAGS += -m32` in `boards/native` and add `LINKFLAGS += -m64` to `boards/native64`.
3. Try to filter the FLAGS in `native64` after the include and replace them with the correct ones.

Reply:
I added a fixup to separate the boards a bit more and remove the `ifeq ($(BOARD),native64)` checks in `boards/native`.

Basically:
- Move `boards/native/{Makefile.features => common_features.inc.mk}` and set `FEATURES_PROVIDED += arch_XXbit` per board.
- Move `boards/native/Makefile.include => makefiles/arch/native.inc.mk` and use `ifneq (,$(filter arch_XXbit,$(FEATURES_PROVIDED))` to set the `LINKFLAGS` & `CFLAGS`.

Creating `makefiles/arch/native.inc.mk` was not strictly necessary here, but as @benpicco said, it makes it a bit more consistent with the other architectures.

Reply:
Fixup needs more work, as it fails:
```
   	Invalid build system patterns found by ./dist/tools/buildsystem_sanity_check/check.sh:
  	Modules should not check the content of FEATURES_PROVIDED/REQUIRED/OPTIONAL:
  		makefiles/arch/native.inc.mk:69:ifneq (,$(filter arch_64bit,$(FEATURES_PROVIDED)))
```